tests/e2e/suite.go

```diff
@@ -153,33 +157,196 @@ func (s *TestSuite) TestQueryBaseFee() {
-
-		gas := int64(1000000)
-		minBaseFee := baseFee.MulInt(math.NewInt(gas))
-
-		// send with the exact expected fee
-		txResp, err := s.SendCoins(
-			ctx,
-			s.user1.KeyName(),
-			s.user1.FormattedAddress(),
-			s.user2.FormattedAddress(),
-			sdk.NewCoins(sdk.NewCoin(cosmosChain.Config().Denom, math.NewInt(10000))),
-			minBaseFee,
-			gas,
-		)
-		s.Require().NoError(err, txResp)
-		s.T().Log(txResp)
+	baseFee := s.QueryBaseFee()
+	gas := int64(20000100)
+	sendAmt := int64(100)
+
+	s.Run("expect fee market fee to increase", func() {
+		s.T().Log("performing sends...")
+		for {
+			// send with the exact expected fee
+			baseFee = s.QueryBaseFee()
+			minBaseFee := baseFee.MulDec(math.LegacyNewDec(gas))[0]
+			// add headroom
+			minBaseFeeCoins := sdk.NewCoins(sdk.NewCoin(minBaseFee.Denom, minBaseFee.Amount.Add(math.LegacyNewDec(10)).TruncateInt()))
+
+			wg := sync.WaitGroup{}
+			wg.Add(3)
+
+			go func() {
+				defer wg.Done()
+				txResp, err := s.SendCoinsMultiBroadcast(
+					context.Background(),
+					s.user1,
+					s.user2,
+					sdk.NewCoins(sdk.NewCoin(cosmosChain.Config().Denom, math.NewInt(sendAmt))),
+					minBaseFeeCoins,
+					gas,
+					400,
+				)
+				s.Require().NoError(err, txResp)
+				s.Require().Equal(uint32(0), txResp.CheckTx.Code, txResp.CheckTx)
+				s.Require().Equal(uint32(0), txResp.TxResult.Code, txResp.TxResult)
+			}()
+
+			go func() {
+				defer wg.Done()
+				txResp, err := s.SendCoinsMultiBroadcast(
+					context.Background(),
+					s.user3,
+					s.user2,
+					sdk.NewCoins(sdk.NewCoin(cosmosChain.Config().Denom, math.NewInt(sendAmt))),
+					minBaseFeeCoins,
+					gas,
+					400,
+				)
+				s.Require().NoError(err, txResp)
+				s.Require().Equal(uint32(0), txResp.CheckTx.Code, txResp.CheckTx)
+				s.Require().Equal(uint32(0), txResp.TxResult.Code, txResp.TxResult)
+			}()
+
+			go func() {
+				defer wg.Done()
+				txResp, err := s.SendCoinsMultiBroadcast(
+					context.Background(),
+					s.user2,
+					s.user1,
+					sdk.NewCoins(sdk.NewCoin(cosmosChain.Config().Denom, math.NewInt(sendAmt))),
+					minBaseFeeCoins,
+					gas,
+					400,
+				)
+				s.Require().NoError(err, txResp)
+				s.Require().Equal(uint32(0), txResp.CheckTx.Code, txResp.CheckTx)
+				s.Require().Equal(uint32(0), txResp.TxResult.Code, txResp.TxResult)
+			}()
+
+			wg.Wait()
+			baseFee = s.QueryBaseFee()
+			s.T().Log("base fee", baseFee.String())
+
+			if baseFee.AmountOf(feemarkettypes.DefaultFeeDenom).GT(math.LegacyNewDec(1000000)) {
+				break
+			}
+		}
+
+		// wait for 5 blocks
+		// query height
+		height, err := s.chain.(*cosmos.CosmosChain).Height(context.Background())
+		s.Require().NoError(err)
+		s.WaitForHeight(s.chain.(*cosmos.CosmosChain), height+5)
+
+		fee := s.QueryBaseFee()
+		s.T().Log("base fee", fee.String())
+
+		amt, err := s.chain.GetBalance(context.Background(), s.user1.FormattedAddress(), baseFee[0].Denom)
+		s.Require().NoError(err)
+		s.Require().True(amt.LT(math.NewInt(initBalance)), amt)
```

Comment:
Looks like we're checking the balance of user1's account? Are we not verifying properties of the fee-state here as well?

Reply:
We are already checking the movement of the fee state in the for loop